core/sis-feature/src/test/java/org/apache/sis/feature/builder/FeatureTypeBuilderTest.java

```diff
@@ -344,4 +347,95 @@ private static void assertPropertyEquals(final String name, final Class<?> value
         }
         assertEquals("valueClass", valueClass, ((DefaultAttributeType<?>) property).getValueClass());
     }
+
+    @Ignore("Obviously a bug.")
+    @Test
+    public void testToStringInternalIsNotNull() {
+        FeatureTypeBuilder featureTypeBuilder = new FeatureTypeBuilder(null);
+        DefaultFeatureType[] defaultFeatureTypeArray = new DefaultFeatureType[6];
+        featureTypeBuilder.setSuperTypes(defaultFeatureTypeArray);
+        StringBuilder stringBuilder = new StringBuilder("");
+        featureTypeBuilder.toStringInternal(stringBuilder);
+
+        assertNotNull(stringBuilder.toString());
+    }
+
+    @Test
+    public void testBuildThrowsRuntimeException() {
```

Comment:
This test modifies directly a package-private field, `FeatureTypeBuilder.identifierCount`, which should be modified only by public API. Then later we get a `CorruptedObjectException`, which is not surprising. This behavior is not part of any public contract, so it should not be tested.